src/js/components/DateInput/DateInput.js

```diff
@@ -152,9 +153,21 @@ const DateInput = forwardRef(
                 // clicking an edge date removes it
                 else if (range) normalizedValue = [nextValue, nextValue];
                 else normalizedValue = nextValue;
+                let nextNormalize = normalize;
+                if (timestamp === undefined) {
```

Comment:
Maybe a brief comment explaining that timestamp will be undefined if no defaultValue or value have been passed in, indicating that we should stay local if the user first picks a date via the Calendar.